### PR TITLE
Button path and ctxId in Auth url

### PR DESCRIPTION
**Pull request by ravikanthreddy89**
Change list:
- "AuthButton" server side route update 
- Added new URL query param for Auth PopUp : "ctxId" 
- Disabled "clientID", "sessionID" and "authSessionID" as query params
- Added new URL query params for AuthButton: "scopes", "responseType", "billingOptions".

### CODE REVIEW

File: src/ui/button/props.js [hunk 1]
```diff
@@ -95,7 +102,11 @@ export type ButtonProps = {|
     clientID : string,
     sessionID : string,
     authButtonSessionID : string,
-    nonce : string
+    nonce : string,
+    scope : string,
```
Comment (ravikanthreddy89) [inline src/ui/button/props.js]: ToDo: Change it to array

File: src/ui/button/props.js [hunk 2]
```diff
@@ -95,7 +102,11 @@ export type ButtonProps = {|
...
+    state? : string | void,
```
Comment (ravikanthreddy89) [inline src/ui/button/props.js]: Remove state as this is not conventional oauth flow

File: src/ui/button/props.js [hunk 3]
```diff
@@ -95,7 +102,11 @@ export type ButtonProps = {|
...
+    responseType : string,
```
Comment (ravikanthreddy89) [inline src/ui/button/props.js]: make it ENUM (code, code+token, code+token+id_token)

File: src/zoid/button/component.jsx [hunk 4]
```diff
@@ -205,7 +205,41 @@ export const getAuthButtonComponent = memoize(() : ZoidComponent<ButtonProps> =>
...
                 }
+            },
+
+            scope: {
+                type:       'string',
+                required:   true,
+                queryParam: true
+            },
+
+            state: {
```
Comment (ravikanthreddy89) [inline src/zoid/button/component.jsx]: remove state

File: src/zoid/button/component.jsx [hunk 5]
```diff
@@ -205,7 +205,41 @@ export const getAuthButtonComponent = memoize(() : ZoidComponent<ButtonProps> =>
                         action: 'auth'
                     };
                 }
+            },
+
+            scope: {
```
Comment (ravikanthreddy89) [inline src/zoid/button/component.jsx]: join the array element and set as value

File: src/zoid/button/component.jsx [hunk 6]
```diff
@@ -205,7 +205,41 @@ export const getAuthButtonComponent = memoize(() : ZoidComponent<ButtonProps> =>
...
+                queryParam: true
+            },
+
+            responseType: {
+                type:       'string',
+                queryParam: true,
+                required:   true
+            },
+
+            billingOptions: {
```
Comment (ravikanthreddy89) [inline src/zoid/button/component.jsx]: use base64 encode the object

File: globals.js [hunk 7]
```diff
@@ -13,8 +13,8 @@ module.exports = {
 
     __PAYPAL_IDENTITY__: {
         __URI__: {
-            __AUTH__:   '/identity/auth',
-            __BUTTON__: '/identity/button'
+            __AUTH__:   '/connect',
+            __BUTTON__: '/apps/button'
```
Comment (ameyanerurkar) [inline globals.js]: can we use /apps/connect

File: test/screenshot/config.js [hunk 8]
```diff
@@ -30,14 +30,25 @@ type ButtonConfig = {|
 
 export const buttonConfigs : Array<ButtonConfig> = [];
 
-buttonConfigs.push({
-    button: {}
-});
+// buttonConfigs.push({
```
Comment (ameyanerurkar) [inline test/screenshot/config.js]: do we need this commented?

File: dist/button.js [hunk 9]
```diff
@@ -1 +1 @@
...
...keyframes show-text {\n        0% { \n    max-width: 0%;\n    opacity: 0;\n    overflow: hidden;\n }\n        100% { \n    max-width: 100%;\n    opacity: 1;\n }\n    }\n\n    "})}function ue(e){le(e)}function ye(e){var n=e.style.color;if(n===Q.BLUE)return d(_,{logoColor:"white"});throw new Error("Unsupported button color: "+n)}function Fe(e){var n=e.style.label;if(n){if(n===X.CONNECT)return d(ne,null,e.content.connectLabel);throw new Error("Unsupported button label: "+(n||"undefined"))}}function Ce(e){var n,r=e.onClick,o=void 0===r?m:r,a=le(e),i=a.fundingSource,f=a.style,s=a.locale,l=a.env,h=a.nonce,p=a.content,c=f.shape,u=f.color,y=function(e,n){e.preventDefault(),e.stopPropagation(),e.target.blur(),o(e,t({fundingSource:i},n))};return d("div",{class:[ee.CONTAINER,ee.SHAPE+"-"+c,ee.ENV+"-"+l].join(" ")},d(ce,{nonce:h,style:f,locale:s}),d("div",t({role:"button"},((n={})["data-button"]=!0,n["data-funding-source"]=i,n),{class:[ee.BUTTON,ee.SHAPE+"-"+c,ee.ENV+"-"+l,ee.COLOR+"-"+u,ee.TEXT_COLOR+"-white"].join(" "),onClick:y,onKeyPress:function(e,n){13!==e.keyCode&&32!==e.keyCode||y(e,n)},tabindex:"0"}),d("div",{class:ee.BUTTON_LABEL},d(ye,{style:f})," ",d(Fe,{style:f,content:p}))))}}]);
```
Comment (bluepnume) [inline dist/button.js]: Can you remove this from the pr. It will be included at publish time.

File: src/zoid/auth/component.jsx [hunk 10]
```diff
@@ -55,21 +55,28 @@ export function getAuthComponent() : ZoidComponent<AuthPropsType> {
...
+                    type:       'string',
+                    queryParam: true,
+                    required:   false
+                },
+
                 clientID: {
                     type:       'string',
                     value:      () => getClientID(),
-                    queryParam: true
+                    queryParam: false
```
Comment (bluepnume) [inline src/zoid/auth/component.jsx]: You can just delete this line entirely, default is false

File: src/zoid/button/component.jsx [hunk 11]
```diff
@@ -205,6 +206,37 @@ export const getAuthButtonComponent = memoize(() : ZoidComponent<ButtonProps> =>
...
+
+            scopes: {
+                type:          'array',
+                required:      true,
+                queryParam:    true,
+                validate:      ({ value }) => {
+                    return validateScopes(value);
+                },
+                queryValue:    ({ value }) => {
+                    return value.join(' ');
```
Comment (bluepnume) [inline src/zoid/button/component.jsx]: Can we do comma separated? Spaces end up being encoded in the url

File: src/zoid/button/component.jsx [hunk 12]
```diff
@@ -205,6 +206,37 @@ export const getAuthButtonComponent = memoize(() : ZoidComponent<ButtonProps> =>
...
+
+            responseType: {
+                type:       'string',
+                queryParam: true,
+                required:   true,
+                validate:   ({ value }) => {
+                    return validateResponseType(value);
+                },
+                queryValue: ({ value }) => {
+                    return value.toLowerCase();
```
Comment (bluepnume) [inline src/zoid/button/component.jsx]: .toLowerCase is unnecessary here, if it's not lowercase validation will fail.

**Comment by bluepnume** [comment]
Looks good! couple of minor comments then will merge 👍

File: src/zoid/button/component.jsx (hunk 11 above)
Comment (vijayeta2387) [reply src/zoid/button/component.jsx]: Space separated would be better as its the same format that is being used in other integrations too

**Pull request merged**